# Introduction
This `markdown` file is used to record the RNA experiment. 

# Drawback of using seedling transcriptome
To use the seedling transcriptome, the seed reproduction will be necessary because we CANNOT keep the seedlings. Thus, a very apperant weakness is this method can apply to the inbred lines or DH lines. Also, the reproduction will require an additional generation, which further limits its usage.

# Cost
## Cost of RNA extraction
- According to the discussion with Sonja on 12.03.2020, she said the cost of RNA extraction per sample could be around 40 Euro to the best of her knowledge (I guess she combined the cost of extraction and library construction).  
- [Digel et al. (2015)](https://doi.org/10.1105/tpc.15.00203) (cited by [Weisweiler et al. (2019)](https://doi.org/10.1186/s12864-019-6174-3)) used RNeasy Micro Kit (Qiagen) and TRIzol (Life Technologies) to extract RNA.   
           - a. The [RNeasy Micro Kit](https://www.qiagen.com/de/products/discovery-and-translational-research/dna-rna-purification/rna-purification/total-rna/rneasy-micro-kit/?clear=true#orderinginformation) costs 508 Euro for 50 preps -> **in average 11 Euro per sample**.   
           - b. The [TRIzol](https://www.thermofisher.com/order/catalog/product/15596026#/15596026) costs 262 and 381 Euro for 100 ml and 200 ml, respectively. 100 ml of TRIzol can used for 100 preps     
      -> **the average cost would be 1.9 Euro** if using 200 ml reagent. (But other reagents are required for TRIzol method. See the table 2 of the [TRIzol user guide](http://tools.thermofisher.com/content/sfs/manuals/trizol_reagent.pdf))  
      -> Prof. Benjamin Stich replied my email on 17.03.2020, and **he said they used [RNeasy Micro Kit](https://www.qiagen.com/de/products/discovery-and-translational-research/dna-rna-purification/rna-purification/total-rna/rneasy-micro-kit/?clear=true#orderinginformation) in their study.**   

- [Pallares et al. 2020](https://www.g3journal.org/content/10/1/143) developed an RNA extraction protocol based on using **[Dynabeads mRNA DIRECT Purification Kit](https://www.thermofisher.com/order/catalog/product/61012#/61012)**, and they reduce the cost to around 1.7 USD per sample. However, this cost was calculated from the method designed for Drosophila, which required 20 micro litter of beads (See [File S2 of Pallares et al. 2020](https://gsajournals.figshare.com/articles/Supplemental_Material_for_Pallares_Picard_and_Ayroles_2020/9905279)). Since our sample is much larger, more beads may needed to collect more RNA.   
The cost of [Dynabeads mRNA DIRECT Purification Kit](https://www.thermofisher.com/order/catalog/product/61012#/61012) is 954 Euro for 10 ml, which can be used for 500 preps if following the protocol of [Pallares et al. 2020](https://www.g3journal.org/content/10/1/143). -> **the average cost per sample is 1.9 Euro... (NOT including other reagents)**  

- **The cost of using the protocol from Sonja's colleague is about 18 Euro per library.**
  
As cost per sample for the methods mentions above is around 2 Euro just for reagent, I think we could to estimate the cost for RNA extraction to be ranging from 4 to 6 Euro per sample.

## Cost of RNA library construction
According [Pallares et al. 2020](https://www.g3journal.org/content/10/1/143), the TM3'-seq method cost 1.5 USD per sample for the reagent. I would like to estimate it to be 2 Euro.

## Cost of RNA sequencing
If we use the 20 milion reads with 2 x 150 bp like [Weisweiler et al. (2019)](https://doi.org/10.1186/s12864-019-6174-3), **the cost would be 60 Euro per sample (10 Euro per Gb)**.


# Pre-trial
## First trial
### 1. Timeline
- Start: 2020-03-13
- Seed germination in the growth chamber with 16 hours of light and 8 hours of darkness under 22&deg;C (from 2020-03-13 to 2020-03-18) 
- End: 2020-03-18 (original plan is to end on 2020-03-20. However, due to the coronavirus, which made things complex and we cannot get liquid nitrogen as our plan, we terminated the trial earlier.)
### 2. Objective
- check the growth of seedlings
- check the feasibility of our experiment system, including sterilization, sowing seeds, harvesting

### 3. Process
- `2020-03-12`: Preparation   
                      - 1. Sterilizing beakers and tweezers with 180&deg;C \~ 200&deg;C.    
                      - 2. Using autoclave to sterilize water (using for germination).      
                      - 3. Cutting filter papers to fit the size of Petri dishes
                      
- `2020-03-13`: Preparation
                      - 1. Using autoclave to sterilize the filter papers. The filter papers wrapped by aluminum foil.  
                      - 2. Using microwave to sterilize water (using to rinse seeds after) -> we use microwave because it is faster than using autoclave.    
                      
- `2020-03-13`: Seed sterilization  
                      - We used the sterilization protocol for wild barley -> **sterilize seeds with 4% hypochlorite with 1 min. But we used the commercial bleach (2.8% sodium hypochlorite without dilution) instead.**  
                      - **All the process of sterilization and sowing seeds were done in the laminar flow cabinet.**  
                      - 1. Preparing bleach and sterile water in separate beakers. And writing the IDs of accessions on the Petri dishes and their lids (Note: Ensure the lids covering Petri dishes when writing the IDs).   
                      - 2. The seeds were sinked the bleach for 1 min and then rinsed in the sterile water  
                      - 3. Putting seeds on a piece of filter paper in a Petri dish.  
                      - 4. Covering seeds with a piece of filter paper  
                      - 5. Adding 9 ml sterile water using the syringe  
                      - 6. Covering Petri dish with the lid and using tapes to secure the lid  
                      - 7. Moving Petri dishes into the growth chamber.
                      - **Time: 3.5 hours for 22 plates (22 accessions with 12 seeds for each)** -> we need to improve this....
                      
- `2020-03-18`: Harvesting seedlings  
                      - Due to the hygiene regulation would start next day, and the liquid nitrogen is not available, so we did not harvest seedlings as original plan
                      - 1. 4 of 22 petri dishes are contaminated -> **the sterilization method should be improved**  
                      - 2. 2 of 22 petri dishes just germinated  
                      - 3. 15 of 22 petri dishes germinated  
                      - 4. the growth of seedling is fast than the expectation -> 5 day is enough  
                      - 5. the seed is difficult to remove from the seedling -> **maybe we need to cut the seedling and its roots? Immersing the whole seedling into liquid nitrogen should make the harvest easier.**  
                      
                      
# Communication with Prof. Stitch
My questions:  
  
1. In your study, the seeds were sterilized by 1% bleach.  
                - Could you please let me know if it means 1% of sodium hypochlorite?   
>> `yes`  
-> **So, we need to dilute the bleach we have, which is 2.8%**  
                - How long did the seeds stay in 1% bleach?   
>> `5 minutes, as far as I recall`  
-> **The time of sterilization need to be longer. This could be the reason of contamination in our first pre-trial.**  
2. When seeds germinating in Petri dishes, were the Petri dishes covered by lids?  
>> `yes, otherwise you need to water very often`  
-> **However, in the first pre-trail, some Petri dishes were dried out, so we may need to seal the Petri dish in the future.**  
  
3. Was the process of sowing, including sterilization and moving seeds into Petri dishes, operated in a laminar flow cabinet or any special environment to reduce the contamination?   
>> `we did that in laminar flow cabinet, but I am rather sure that it is not required.`   
-> **I can try to sow the seeds out of laminar flow cabinet in the next pre-trail. If it is really not required, the workflow could be much faster.**  
Or, is it not necessary to be very cautious about the microbes in the air?   
>> `depends on your environment`  

  
4. In your study, the seedlings were collected five days after germination.  
Did you use a specific standard to select seedlings when the germination speed is uneven between samples?   
>> `no, for us that was very homogenous. If you have doubts for your material, you could make the harvest stage dependent.`  
-> According to our first pre-trail, the growth is heterogeneous...

5. For the RNA extraction, you cited the work of Digel et al. (2015).  
However, Digel et al. used the RNeasy Micro Kit (Qiagen) and TRIzol (Life Technologies) for the RNA extraction.  

Could you please let me know which one was used in your study?  
>> `the first one`


6. Was the mRNA extracted from whole seedlings including the attached seeds, or the seeds were removed during harvesting seedlings?  
>> `the seeds were removed`  




# Growth condition of seedlings
## Issue
- (1) When should we harvest seedlings?  
-> According to our first pre-trial, the five days are enough (seven days will be too many), but the growth of seedlings is not homogeneous (the reason is still not clear.)
(EDITING)


# RNA extraction
[Weisweiler et al. (2019)](https://doi.org/10.1186/s12864-019-6174-3) cited the work of [Digel et al. (2015)](https://doi.org/10.1105/tpc.15.00203) for the details of RNA extraction. Here is the description of [Digel et al. (2015)](https://doi.org/10.1105/tpc.15.00203): 
> The samples harvested for RNA extraction were frozen immediately in liquid nitrogen and stored at −80°C. Total RNA, excluding miRNAs, was extracted from ground tissue using an **RNeasy Micro Kit (Qiagen) and TRIzol (Life Technologies) for RNA sequencing and qRT-PCR, respectively**. Residual DNA was removed using a **DNA-free kit (Ambion).** RNA extraction and DNase treatment were performed according to the manufacturer’s instructions. The RNA concentration and integrity were determined using a 2100 Bioanalyzer (Agilent) before RNA library preparation for RNA sequencing. Owing to the column-based RNA purification method of the RNeasy Micro Kit, only transcript species larger than 200 nucleotides were further processed for RNA sequencing.  


# RNA sequencing
## Sequence depth
According to [Weisweiler et al. (2019)](https://doi.org/10.1186/s12864-019-6174-3), they had around 20 million 2x150 bp reads for one sample. Their finding suggested using 1% \~ 5% of reads can still obtain acceptable prediction ability. 
  
In their article, the authors mentioned that the cost of 20 million 2x150 bp reads is 60 Euro and the library construction is 2 Euro per sample if using the latest protocol (BRB-seq; [Alpern et al. 2019](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1671-x)). (According to [Pallares et al. 2020](https://www.g3journal.org/content/ggg/10/1/143.full.pdf), the commercial RNA kits cost 35 USD \~ 44 USD per sample.)   

> However, the cost of genotyping one sample with the barley 50K SNParray is with about 50 Euro \[42] also less than the mRNA sequencing analysis. When generating it newly with latest protocols and sequencing chemistry one could expect that the mRNA sequencing of **one sample would cost about 2 Euro for the mRNA library preparation \[43] as well as 60 Euro for 20 million 2x150 bp reads**.  

If we calculate the cost according to this number, the total cost will be `(800 + 15 parents * 3 batches) * 62 = 52390 Euro`. This is definitely not possible to spend so much money on sequencing. So, we should use other methods for sequencing RNA.
  
# solution 1: reduce depth  
If we take the authors suggestion reducing the read number to 2 million 2x150 bp reads (10% * 20 million 2x150 bp) the cost per sample will be `(800 + 15*3) * [(60 * 0.1) + 2] = 6760 Euro` which is more affordable.  
- Drawback: by reducing the read depth, the resulted data may not informative enough for establishment of prediction models.   
-> **The finding of [Weisweiler et al. (2019)](https://doi.org/10.1186/s12864-019-6174-3) has not been proved as a universal result that can apply to any population and any trait, so Katharina reminded me that I should not design the experiment by assuming the lower read depth can also work well in my project.

# solution 2: reduce sample size
`10000 euro / 62 = 161 libraries`  
- Drawback: by reduce the sample size to less than 25% of original sample size, we can expect the accuracy of model will decline and also the conclusion obtained from this model will be less general.  
## Sample selection
If we choose to reduce the sample size, then the next question will be how to select sub-sample. This question depent on the study objective:  
- study the effect of crosses, such as sampling mainly from a population with the wild parent and a population without the wild parent -> **we may sample about 4 populaitons with 40 lines for each or sampling 100 lines from a single elite line population and 60 lines for 1 or 2 other populations.**  
- study the effect of diverse genetic background -> **we may equally sample about 10 lines from 14 populations.**  
- **NOTE**: We need to check the metabolic data and the phenotypic data to prevent wasting money on sequencing for the lines without complete data.
  
## RNA library
We currently have two new methods for preparing RNA library.
- TM3'seq: [Pallares et al. 2020](https://www.g3journal.org/content/ggg/10/1/143.full.pdf) -> We will use this one
- BRB-seq: [Alpern et al. 2019](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1671-x)
